### PR TITLE
override ap-normalize version with  0.1.0.1

### PR DESCRIPTION
**Pull request by yaitskov**
sets ap-normalize with fixed bug causing failure on Mac OS Big Sur.

### CODE REVIEW

File: pkgs/development/haskell-modules/non-hackage-packages.nix [hunk 1]
```diff
@@ -47,5 +47,7 @@ self: super: {
   # Generated with:
   # nix-shell -I nixpkgs=$PWD -p cabal-install -p cabal2nix --run 'cabal update; cabal2nix cabal://arion-compose > pkgs/applications/virtualization/arion/arion-compose.nix'
   arion-compose = self.callPackage ../../applications/virtualization/arion/arion-compose.nix {};
+  # cabal2nix --revision <rev> https://gitlab.com/lysxia/ap-normalize.git
```
Comment (sternenseemann) [inline pkgs/development/haskell-modules/non-hackage-packages.nix:50]: Rather use the following for generating the package which gets around the use of fetchgit (which is slow) and uses the sdist so we are closer to the hackage package.

```suggestion
  # cabal2nix cabal://ap-normalize-$version
```

File: pkgs/development/misc/haskell/ap-normalize/default.nix [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  src = fetchgit {
+    url = "https://gitlab.com/lysxia/ap-normalize.git";
+    sha256 = "0xfr00dy5qiisiwkhbmmisbv3zmfak5i8fx3b9ymxswxwvvbhjnc";
+    rev = "ed521b63f02f1fe7d4e4ec5f8c5de2d4f76ab017";
+    fetchSubmodules = true;
+  };
+  libraryHaskellDepends = [ base ];
+  testHaskellDepends = [ base inspection-testing transformers ];
+  description = "Self-normalizing applicative expressions";
+  license = stdenv.lib.licenses.mit;
```
Comment (sternenseemann) [inline pkgs/development/misc/haskell/ap-normalize/default.nix:16]: Try upgrading your version of `cabal2nix`, it shouldn't generate `stdenv.lib` attributes anymore (which has been deprecated).

**Review by sternenseemann — CHANGES_REQUESTED** [review]
LGTM aside from comments. Also make sure your commit message follows the [Contributing guideline](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes).

File: pkgs/development/haskell-modules/non-hackage-packages.nix (hunk 1 above)
Comment (yaitskov) [reply pkgs/development/haskell-modules/non-hackage-packages.nix]: fixed

File: pkgs/development/misc/haskell/ap-normalize/default.nix (hunk 2 above)
Comment (yaitskov) [reply pkgs/development/misc/haskell/ap-normalize/default.nix]: I replaced stdenv with just lib. The warning disappeared.

File: pkgs/development/haskell-modules/non-hackage-packages.nix (hunk 1 above)
Comment (sternenseemann) [reply pkgs/development/haskell-modules/non-hackage-packages.nix]: Can you also regenerate with that command (and ideally a newer version of cabal2nix which you should be able to get from the darwin 21.05 channel)

**Comment by yaitskov** [comment]
commits have been squashed for better commit message.

File: pkgs/development/misc/haskell/ap-normalize/default.nix [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+, transformers
+}:
+mkDerivation {
+  pname = "ap-normalize";
+  version = "0.1.0.1";
+  src = fetchgit {
+    url = "https://gitlab.com/lysxia/ap-normalize.git";
+    sha256 = "0xfr00dy5qiisiwkhbmmisbv3zmfak5i8fx3b9ymxswxwvvbhjnc";
+    rev = "ed521b63f02f1fe7d4e4ec5f8c5de2d4f76ab017";
+    fetchSubmodules = true;
```
Comment (SuperSandro2000) [inline pkgs/development/misc/haskell/ap-normalize/default.nix:11]: The repo contains no submodules.

File: pkgs/development/misc/haskell/ap-normalize/default.nix [hunk 4]
```diff
@@ -0,0 +1,17 @@
+{ mkDerivation, base, fetchgit, inspection-testing, lib
+, transformers
+}:
+mkDerivation {
+  pname = "ap-normalize";
+  version = "0.1.0.1";
+  src = fetchgit {
```
Comment (SuperSandro2000) [inline pkgs/development/misc/haskell/ap-normalize/default.nix:7]: Lets use fetchFromGitLab.

Comment (sternenseemann) [reply pkgs/development/misc/haskell/ap-normalize/default.nix:7]: As said [above](https://github.com/NixOS/nixpkgs/pull/130946#discussion_r674075476), regenerating the file using `cabal2nix` will cause it to fetch the sdist tarball from hackage, which is preferrable in every way.

**Comment by sternenseemann** [comment]
@ofborg build haskellPackages.ap-normalize

**Comment by sternenseemann** [comment]
Sorry, messed the commit message up, didn't realize GitHub always would put the PR title up top :/

**Pull request merged**